CONTRIBUTING.md

```diff
@@ -1,5 +1,96 @@
+adapt to the change. The idea behind release documentation is that when a naive
+user with limited internal knowledge of the project is upgrading from the
+previous release to the new one, they should be able to read the release notes,
+understand if they need to update their program which uses qiskit-ibm-transpiler,
+and how they would go about doing that. It ideally should explain why
+they need to make this change too, to provide the necessary context.
+
+To make sure we don't forget a release note or if the details of user facing
+changes over a release cycle we require that all user facing changes include
+documentation at the same time as the code. To accomplish this, we use the
+[Towncrier](https://towncrier.readthedocs.io/en/stable/) tool.
+
+### Adding a new release note
+
+To create a new release note, first find either the issue or PR number associated with
+your change from GitHub because Towncrier links every release note to a GitHub issue
+or PR. If there is no associated issue and you haven't yet opened up the PR so you don't
+yet have a PR number, you can use the value `todo` at first, then go back and rename the
+file once you open up the PR and have its number.
+
+Then, identify which type of change your release note is:
+
+- `feat` (new feature)
+- `upgrade` (upgrade note)
+- `deprecation` (deprecation)
+- `bug` (bug fix)
+- `other` (other note)
+
+Now, create a new file in the `release-notes/unreleased` folder in the format `<github-number>.<type>.rst`,
+such as `156.bug.rst` or `231.feat.rst`.
+
+Open up the new release note file and provide a description of the change, such as what users need
+to do. The files use RST syntax and you can use mechanisms like code blocks and cross-references.
+
+Example notes:
+
+```rst
+Add `dd_barrier` optional input to
+:class:`.PadDynamicalDecoupling`
+constructor to identify portions of the circuit to apply dynamical
+decoupling (dd) on selectively. If this string is contained in the
+label of a barrier in the circuit, dd is applied on the delays ending
+with it (on the same qubits); otherwise, it is not applied.
+```
+
+```
+When a single backend is retrieved with the `instance` parameter,
+
+.. code:: python
+
+  service.backend('ibm_torino', instance='ibm-q/open/main')
+  # raises error if torino is not in ibm-q/open/main but in a different instance
+  # the user has access to
+  service = QiskitRuntimeService(channel="ibm_quantum", instance="ibm-q/open/main")
+  service.backend('ibm_torino') # raises the same error
+
+if the backend is not in the instance, but in a different one the user
+has access to, an error will be raised. The same error will now be
+raised if an instance is passed in at initialization and then a
+backend not in that instance is retrieved.
+```
+
+In general, you want the release notes to include as much detail as
+needed so that users will understand what has changed, why it changed, and how
+they'll have to update their code.
```

Comment:
```suggestion
In general, you want the release notes to include as much detail as needed so that users will understand what has changed, why it changed, and how they'll have to update their code.
```